contracts/Aavegotchi/facets/ERC1155BuyOrderFacet.sol

```diff
@@ -0,0 +1,256 @@
+import {LibERC20} from "../../shared/libraries/LibERC20.sol";
+import {LibERC1155} from "../../shared/libraries/LibERC1155.sol";
+import {LibItems} from "../libraries/LibItems.sol";
+import {IERC20} from "../../shared/interfaces/IERC20.sol";
+import {IERC1155} from "../../shared/interfaces/IERC1155.sol";
+import {LibMeta} from "../../shared/libraries/LibMeta.sol";
+import {Modifiers, ERC1155BuyOrder} from "../libraries/LibAppStorage.sol";
+import {BaazaarSplit, LibSharedMarketplace, SplitAddresses} from "../libraries/LibSharedMarketplace.sol";
+import "../WearableDiamond/interfaces/IEventHandlerFacet.sol";
+
+contract ERC1155BuyOrderFacet is Modifiers {
+    event ERC1155BuyOrderAdd(
+        uint256 indexed buyOrderId,
+        address indexed buyer,
+        address erc1155TokenAddress,
+        uint256 erc1155TokenId,
+        uint256 indexed category,
+        uint256 priceInWei,
+        uint256 quantity,
+        uint256 duration,
+        uint256 time
+    );
+    event ERC1155BuyOrderUpdate(uint256 indexed buyOrderId, uint256 indexed category, uint256 priceInWei, uint256 quantity, uint256 time);
+    event ERC1155BuyOrderCancel(uint256 indexed buyOrderId, uint256 time);
+    event ERC1155BuyOrderExecute(
+        uint256 indexed buyOrderId,
+        address indexed buyer,
+        address seller,
+        address erc1155TokenAddress,
+        uint256 erc1155TokenId,
+        uint256 indexed category,
+        uint256 priceInWei,
+        uint256 quantity,
+        uint256 time
+    );
+
+    function getERC1155BuyOrder(uint256 _buyOrderId) external view returns (ERC1155BuyOrder memory buyOrder_) {
+        buyOrder_ = s.erc1155BuyOrders[_buyOrderId];
+        require(buyOrder_.timeCreated != 0, "ERC1155BuyOrder: ERC1155 buyOrder does not exist");
+    }
+
+    function getERC1155BuyOrderIdsByTokenId(
+        address _erc1155TokenAddress,
+        uint256 _erc1155TokenId
+    ) external view returns (uint256[] memory buyOrderIds_) {
+        buyOrderIds_ = s.erc1155TokenToBuyOrderIds[_erc1155TokenAddress][_erc1155TokenId];
+    }
+
+    function getERC1155BuyOrdersByTokenId(
+        address _erc1155TokenAddress,
+        uint256 _erc1155TokenId
+    ) external view returns (ERC1155BuyOrder[] memory buyOrders_) {
+        uint256[] memory buyOrderIds = s.erc1155TokenToBuyOrderIds[_erc1155TokenAddress][_erc1155TokenId];
+        uint256 length = buyOrderIds.length;
+        buyOrders_ = new ERC1155BuyOrder[](length);
+        for (uint256 i; i < length; i++) {
+            buyOrders_[i] = s.erc1155BuyOrders[buyOrderIds[i]];
+        }
+    }
+
+    function placeERC1155BuyOrder(
+        address _erc1155TokenAddress,
+        uint256 _erc1155TokenId,
+        uint256 _priceInWei,
+        uint256 _quantity,
+        uint256 _duration
+    ) external {
+        uint256 cost = _quantity * _priceInWei;
+        require(cost >= 1e18, "ERC1155BuyOrder: cost should be 1 GHST or larger");
```

Comment:
Let's change this to 0.001 GHST @orionstardust

Reply:
Updated.